contracts/ZetaChain.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "./OnlySystem.sol";
+import "./BytesHelperLib.sol";
+import "./SwapHelperLib.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/SystemContract.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/IWZETA.sol";
+
+abstract contract ZetaChain is zContract, OnlySystem {
+    SystemContract public systemContract;
```

Comment:
you inherit from OnlySystem the system contract address; looks redundant to have this one also.

Reply:
I think we need this variable to able to use it from other functions without having to pass it explicitly, for example:

https://github.com/zeta-chain/toolkit/pull/135

Reply:
`OnlySystem` only has system contract address, `systemContract` on the other hand is a reference to the system contract, which then may be used in other functions like so https://github.com/zeta-chain/toolkit/pull/135

Open to suggestions on how to streamline this.

Reply:
I suppose we can do

```solidity
abstract contract ZetaChain is zContract, OnlySystem {
    SystemContract public systemContract;
    mapping(uint256 => address) private systemContracts;

    constructor() OnlySystem(_initializeSystemContracts()) {
        systemContract = SystemContract(systemContractAddress);
    }

    function _initializeSystemContracts() private returns (address) {
        systemContracts[7000] = 0x91d18e54DAf4F677cB28167158d6dd21F6aB3921;
        systemContracts[7001] = 0xEdf1c3275d13489aCdC6cD6eD246E72458B8795B;
        return systemContracts[block.chainid];
    }
}
```

Reply:
The example of why I think we need a reference to the system contract is:

```solidity
abstract contract ZetaChain is zContract, OnlySystem {
    SystemContract public systemContract;
    mapping(uint256 => address) private systemContracts;

    constructor() OnlySystem(_initializeSystemContracts()) {
        systemContract = SystemContract(systemContractAddress);
    }

    function _initializeSystemContracts() private returns (address) {
        systemContracts[7000] = 0x91d18e54DAf4F677cB28167158d6dd21F6aB3921;
        systemContracts[7001] = 0xEdf1c3275d13489aCdC6cD6eD246E72458B8795B;
        return systemContracts[block.chainid];
    }

    function doSomething() public {
        // systemContract.uniswapv2Router02Address()
    }
}
```

Reply:
then let's reference the contract in OnlySystem. I don't like to use two memory slot for exactly the same data.

Reply:
Moved all the system contract logic to `OnlySystem`: d4ecf31